aframe/tasks/infer/infer.py

```diff
@@ -162,37 +166,105 @@ def foreground_files(self):
             [Path(targets["foreground"].path) for targets in self.targets]
         )
 
-    @classmethod
-    def get_shifts(cls, files: List[Path]):
-        shifts = []
-        for f in files:
-            with h5py.File(f) as f:
-                shift = f["parameters"]["shift"][0]
-                shifts.append(shift)
-        return shifts
+    @property
+    def metadata_files(self):
+        return np.array(
+            [Path(targets["metadata"].path) for targets in self.targets]
+        )
+
+    @property
+    def timeseries_files(self):
+        if self.return_timeseries:
+            return np.array(
+                [Path(targets["timeseries"].path) for targets in self.targets]
+            )
+        return None
+
+    def get_metadata(self):
+        files = self.metadata_files
+        num_files = len(files)
+        background_lengths = np.zeros(num_files)
+        foreground_lengths = np.zeros(num_files)
+        shifts = np.zeros((num_files, len(self.shifts)))
+        for i, f in enumerate(files):
+            with open(f, "r") as f:
+                data = json.load(f)
+            background_lengths[i] = data["background_length"]
+            foreground_lengths[i] = data["foreground_length"]
+            shifts[i] = data["shifts"]
+
+        return background_lengths, foreground_lengths, shifts
+
+    def aggregate_timeseries(self):
+        index = []
+        with h5py.File(self.timeseries_output, "w") as f:
+            ts_group = f.create_group("timeseries")
+            for ts in self.timeseries_files:
+                with h5py.File(ts, "r") as g:
+                    t0 = g.attrs["t0"]
+                    shifts = g.attrs["shifts"]
+                    background = g["background"][:]
+                    foreground = g["foreground"][:]
+                    ts_id = f"{t0}_{shifts}"
+                    subgroup = ts_group.create_group(ts_id)
+                    subgroup.create_dataset("background", data=background)
+                    subgroup.create_dataset("foreground", data=foreground)
+                    index.append((t0, shifts, f"/timeseries/{ts_id}"))
+
+            # Create an index to make it easier to look up
+            # specific segments and shifts
+            dtype = np.dtype(
+                [
+                    ("t0", np.float64),
+                    ("shifts", np.int32, (len(shifts),)),
+                    ("path", h5py.string_dtype(encoding="utf-8")),
+                ]
+            )
+            index_array = np.array(index, dtype=dtype)
+            f.create_dataset("index", data=index_array)
```

Comment:
This works, but it's not a great solution - still awkward to get to the data you want